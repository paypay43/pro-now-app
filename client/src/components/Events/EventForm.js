import React, { useState } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import AuthApiService from '../../services/auth-api-service';
import EventApiService from '../../services/event-api-service';
import EventLocationInput from './EventLocationInput';
import DatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';
import './event.css';
import PlacesAutocomplete, {
  geocodeByAddress,
  getLatLng
} from 'react-places-autocomplete';

export default function EventForm(props) {
  const [error, setError] = useState(null);
  const [date, setDate] = useState(null);
  const [place, setPlace] = useState('');

  /*const handleSubmitJwtAuth = ev => {
    ev.preventDefault();
    ;
  };*/

  const submitForm = e => {
    e.preventDefault();
    setError(null);
    const { event_name, description } = e.target;

    EventApiService.postEvent({
      event_name: event_name.value,
      location: place,
      description: description.value,
      date: date
    })
      .then(event => {
        event_name.value = '';
        description.value = '';
        props.create(event.id);
      })
      .catch(res => {
        setError({ error: res.error });
      });
  };

  const displayError = () => {
    return Object.keys(error).map(e => {
      return (
        <span key={e}>
          {e}: {error[e]}
        </span>
      );
    });
  };

  const changeView = () => {
    props.changeView();
  };

  const changeLocation = value => {
    setPlace(value);
  };

  return (
    <form onSubmit={submitForm} className="event-form app-form">
      <h2>Create Event</h2>
      <div role="alert">{error && <p className="red">{displayError()}</p>}</div>
      <label className="label-control" htmlFor="event_name">
        Event Name
      </label>
      <input
        name="event_name"
        type="text"
        className="form-control"
        required
        autoFocus=""
      />
      <label className="label-control" htmlFor="location">
        Location
      </label>
      <EventLocationInput setAddress={changeLocation} />
      <label className="label-control" htmlFor="decsription">
        Description
      </label>
      <textarea
        name="description"
        type="text"
        className="form-control"
        required
      />
      <label className="label-control" htmlFor="date">
        Date/Time
      </label>
      <DatePicker selected={date} onChange={value => setDate(value)} />
      <button className="btn add" type="submit">
        Add
      </button>
    </form>
  );
}
