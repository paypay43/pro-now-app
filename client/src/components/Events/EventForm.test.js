import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import EventForm from './eventForm';

describe(`EventForm component`, () => {
  /*it('renders loginform', () => {
    const wrapper = shallow(<EventForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });*/

  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<EventForm />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
