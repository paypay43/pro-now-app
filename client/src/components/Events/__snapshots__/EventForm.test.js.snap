// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventForm component renders loginform 1`] = `
<form
  className="event-form app-form"
  onSubmit={[Function]}
>
  <h2>
    Create Event
  </h2>
  <div
    role="alert"
  />
  <label
    className="label-control"
    htmlFor="event_name"
  >
    Event Name
  </label>
  <input
    autoFocus=""
    className="form-control"
    name="event_name"
    required={true}
    type="text"
  />
  <label
    className="label-control"
    htmlFor="location"
  >
    Location
  </label>
  <EventLocationInput
    setAddress={[Function]}
  />
  <label
    className="label-control"
    htmlFor="decsription"
  >
    Description
  </label>
  <textarea
    className="form-control"
    name="description"
    required={true}
    type="text"
  />
  <label
    className="label-control"
    htmlFor="date"
  >
    Date/Time
  </label>
  <DatePicker
    allowSameDay={false}
    dateFormat="MM/dd/yyyy"
    dateFormatCalendar="LLLL yyyy"
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="scroll"
    inlineFocusSelectedMonth={false}
    monthsShown={1}
    nextMonthButtonLabel="Next month"
    onBlur={[Function]}
    onChange={[Function]}
    onClickOutside={[Function]}
    onFocus={[Function]}
    onInputClick={[Function]}
    onInputError={[Function]}
    onKeyDown={[Function]}
    onMonthChange={[Function]}
    onSelect={[Function]}
    onYearChange={[Function]}
    preventOpenOnFocus={false}
    previousMonthButtonLabel="Previous Month"
    readOnly={false}
    renderDayContents={[Function]}
    selected={null}
    shouldCloseOnSelect={true}
    showTimeInput={false}
    showTimeSelect={false}
    strictParsing={false}
    timeCaption="Time"
    timeInputLabel="Time"
    timeIntervals={30}
    withPortal={false}
  />
  <button
    className="btn add"
    type="submit"
  >
    Add
  </button>
</form>
`;
